In [10]:
from sklearn.preprocessing import MinMaxScaler
from sklearn.preprocessing import StandardScaler
import pandas as pd

In [8]:
data = pd.DataFrame({
    'age': [20,50,100],
    'income': [20000,50000,100000]
})

scaler = MinMaxScaler()

normalize_data = scaler.fit_transform(data)
df_normal = pd.DataFrame(normalize_data,columns=['age','income'])
df_normal

age  income
0  0.000   0.000
1  0.375   0.375
2  1.000   1.000

- Why do we normalize data
    - easier to calculate
    - 
- a lot of preferences to salary over age in the example
- our model only use number

- 100-20 = 80
- 50-20 = 30
- 30/80 = 0.375


In [9]:
data

age  income
0   20   20000
1   50   50000
2  100  100000

In [ ]:
std_scaler =StandardScaler()        # its give z-score transformation for each plot point

In [16]:
std_data = std_scaler.fit_transform(data)
std_data

array([[-1.1111678 , -1.1111678 ],
       [-0.20203051, -0.20203051],
       [ 1.31319831,  1.31319831]])

# titanic